### PR TITLE
Save diagram layout : Fix #213

### LINKED ISSUE #213: Save the current diagram layout
### Prerequisites

- [x] I have written a descriptive issue title
- [x] I have verified that I am running the latest version of the UI-DSM
- [x] I have searched [open](https://github.com/RHEAGROUP/UI-DSM/issues) and [closed](https://github.com/RHEAGROUP/UI-DSM/issues?q=is%3Aissue+is%3Aclosed) issues to ensure it has not already been reported

### Description

Inside the PhysicalFlowView, the user should be able to save the current layout of the diagram inside the server, as a JSON file. 
This file should store all information regarding the position of each node and the represented 10-25 Thing (reference to the iid).
The configuration file should be linked a specific Task (store all configuration related to a Task into a folder named by the id of the Task).
### System Configuration

- UI-DSM version:
  - [ ] UI-DSM:         
  - [ ] Other:              
- Environment (Operating system, version and so on):
- .NET Framework version:
- Additional information:

### PR DESCRIPTION
**Pull request by nabilatrhea**
### Prerequisites

- [x] I have written a descriptive pull-request title
- [x] I have verified that there are no overlapping [pull-requests](https://github.com/RHEAGROUP/UI-DSM/pulls) open
- [x] I have verified that I am following the UI-DSM [code style guidelines](https://raw.githubusercontent.com/RHEAGROUP/UI-DSM/master/.github/CONTRIBUTING.md)
- [x] I have provided test coverage for my change (where applicable)

### Description

Fix #213 

A participant of a project is able to save a diagram layout that he define

### CODE REVIEW

File: UI_DSM.Client/Components/NormalUser/Views/PhysicalFlowView.razor.cs [hunk 1]
```diff
@@ -144,6 +150,9 @@ protected override void OnInitialized()
 
             this.Diagram.MouseUp += this.Diagram_MouseUp;
             this.Diagram.MouseDoubleClick += this.Diagram_MouseDoubleClick;
+
+            this.disposables.Add(this.WhenAnyValue(x => x.ViewModel.IsOnSavingMode)
```
Comment (antoineatrhea) [inline UI_DSM.Client/Components/NormalUser/Views/PhysicalFlowView.razor.cs:154]: The collection is not dispose, please add `this.disposables.ForEach(x => x.Dispose());` inside the Dispose Method

File: UI_DSM.Client/Model/DiagramNode.cs [hunk 2]
```diff
@@ -20,6 +20,9 @@ namespace UI_DSM.Client.Model
     /// </summary>
     public class DiagramNode : NodeModel, IDiagramModel
     {
+
+        public Guid ThingId { get; set; }
```
Comment (antoineatrhea) [inline UI_DSM.Client/Model/DiagramNode.cs:24]: Missing documentation

File: UI_DSM.Server/Modules/DiagrammingConfigurationModule.cs [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+            app.MapPost($"{this.MainRoute}/Save", this.SaveLayoutConfiguration)
+                .Accepts<List<DiagramLayoutInformationDto>>("application/json")
+                .WithTags("Layout")
+                .WithName("Layout/Save");
+        }
+
+        /// <summary>
+        ///     Saves the product diagram layout
+        /// </summary>
+        /// <param name="context">The <see cref="HttpContext" /></param>
```
Comment (antoineatrhea) [inline UI_DSM.Server/Modules/DiagrammingConfigurationModule.cs:65]: Missing documentation for parameter. Please also use the "string" instead of "String"

**Review by antoineatrhea — CHANGES_REQUESTED** [review]
Small changes needed

**Review by alexatrhea — APPROVED** [review]

**Review by antoineatrhea — APPROVED** [review]

**Pull request merged**